internal/codegen/golang/gen.go

```diff
@@ -288,3 +293,42 @@ func usesBatch(queries []Query) bool {
 	}
 	return false
 }
+
+func filterUnusedStructs(enums []Enum, structs []Struct, queries []Query) ([]Enum, []Struct) {
+	keepTypes := make(map[string]struct{})
+
+	for _, query := range queries {
+		if !query.Arg.isEmpty() {
+			keepTypes[strings.TrimPrefix(query.Arg.Type(), "Null")] = struct{}{}
+			if query.Arg.IsStruct() {
+				for _, field := range query.Arg.Struct.Fields {
+					keepTypes[strings.TrimPrefix(field.Type, "Null")] = struct{}{}
+				}
+			}
+		}
+		if query.hasRetType() {
+			keepTypes[strings.TrimPrefix(query.Ret.Type(), "Null")] = struct{}{}
+			if query.Ret.IsStruct() {
+				for _, field := range query.Ret.Struct.Fields {
+					keepTypes[strings.TrimPrefix(field.Type, "Null")] = struct{}{}
+				}
+			}
+		}
+	}
```

Comment:
The trimming behavior here has a subtle bug. Type names come from user-defined table names. A table named `null_foo` will have a struct named `NullFoo`. Instead of trimming the null prefix, The enum code below should check for any use of the enum struct or the nullable enum struct.

Reply:
Good point, will update